### PR TITLE
Auto delete MetaData when = 0

### PR DESCRIPTION
**Pull request by Lejo1**
We still automatic delete a key of a metadata or mod_storage data if it is = ""
This PR also auto deletes if it is = 0
As we basically also return 0 for no value this makes no compatibility problems
This saves memory when mods for example remove a player and set the var to default = 0

## How to test
``` Lua
local storage = minetest.get_mod_storage()

storage:set_string("placeholder", "make sure file is not null")

storage:set_int("int", 100)
storage:set_int("int", 0)
storage:set_float("float", 100.1)
storage:set_float("float", 0)

storage:set_float("float2", 100.1)
storage:set_float("float2", 0.0)

storage:set_string("string", "stringggggggggggg")
storage:set_string("string", "")
```
All the entries will disappear without the PR float, float2 and int will stay = 0

### CODE REVIEW

File: src/script/lua_api/l_metadata.cpp [hunk 1]
```diff
@@ -153,7 +153,9 @@ int MetaDataRef::l_set_int(lua_State *L)
 	MetaDataRef *ref = checkobject(L, 1);
 	std::string name = luaL_checkstring(L, 2);
 	int a = luaL_checkint(L, 3);
-	std::string str = itos(a);
+	std::string str = std::string("");
```
Comment (SmallJoker) [inline src/script/lua_api/l_metadata.cpp]: ` = std::string("")` is superfluous.

File: src/script/lua_api/l_metadata.cpp [hunk 2]
```diff
@@ -191,7 +193,9 @@ int MetaDataRef::l_set_float(lua_State *L)
 	MetaDataRef *ref = checkobject(L, 1);
 	std::string name = luaL_checkstring(L, 2);
 	float a = readParam<float>(L, 3);
-	std::string str = ftos(a);
+	std::string str = std::string("");
```
Comment (SmallJoker) [inline src/script/lua_api/l_metadata.cpp]: Ditto

**Comment by SmallJoker** [comment]
The only issue I can see is when you mix `MetaRef:get()` and `MetaRef:set_int()`. Latter will write `"0"`, but former will return `nil`. In terms of storage optimization this is totally fine.

**Comment by Lejo1** [comment]
In my case it’s a money mod. Where I only use set_int and get_int.
When creating a function to fully reset a play including all data of all
mods I came up to this.

**Comment by Moth-Tolias** [comment]
what's the status on this?

**Pull request merged**